File: c++/cook.cpp
#include "cook.h"

#include <string>
#include <iostream>
#include <list>

using namespace std;

Cook::Cook(string _name, int _age, int _wage): Worker(_name, _age, _wage) {
    order = list<string>();
    food = list<string>();
}

Cook::~Cook() {
    cout << "Instance removed." << endl;
}

void Cook::recieve_checked_order(list<string> orders) {
    this -> order = orders;
    cout << this -> name << "recieved orders" << endl;
    this -> cook_food();
}

void Cook::cook_food() {
    list<string> foods;
    foods = this -> order;
    this -> order.clear();
    cout << "cook" << endl;
    this -> food = foods;
}

void Cook::send_food(Server server) {
    //list<string> foods;
    //foods = this -> food;
    // cout << "sended" << foods << "to" << server.name << endl;
    //std::string joined = boost::algorithm::join(foods, ",");
    server.recieve_checked_food();
    this -> food.clear();
}

File: c#/Cook.cs
using System;
using System.Collections.Generic;

public class Cook:Worker {
    public List<string> order;
    public List<string> food;

    public Cook(string _name, int _age, int _wage): base(_name, _age, _wage) {
        order = new List<string>();
        food = new List<string>();
    }

    ~Cook() {
        System.Console.WriteLine("Instance removed.");
    }

    public void recieve_checked_order(List<string> orders){
        this.order = orders;
        System.Console.WriteLine(this.name + "recieved orders");
        this.cook_food();
    }

    public void cook_food(){
        List<string> foods = this.order;
        this.order.Clear();
        System.Console.WriteLine("cook");
        this.food = foods;
    }

    public void send_food(Server server) {
        List<string>foods = this.food;
        System.Console.WriteLine("sended " + foods + " to " + server.name);
        server.recieve_checked_food(String.Join(", ", foods.ToArray()));
        this.food.Clear();
    }
}
